apps/wing-console/console/ui/src/layout/default-layout.tsx

```diff
@@ -367,27 +410,42 @@ export const DefaultLayout = ({
                     className={classNames(
                       theme.border4,
                       "flex-shrink w-80 min-w-[10rem] z-10",
+                      USE_EXTERNAL_THEME_COLOR,
                     )}
                   >
                     <div
                       className={classNames(
                         "w-full h-full relative",
-                        USE_EXTERNAL_THEME_COLOR,
-                        theme.bg4,
+                        theme.bg3,
                         layout.panels?.rounded && "rounded-lg overflow-hidden",
                       )}
                     >
-                      {metadata.data && (
-                        <ResourceMetadata
-                          node={metadata.data.node}
-                          inbound={metadata.data.inbound}
-                          outbound={metadata.data.outbound}
-                          onConnectionNodeClick={(path) => {
-                            expand(path);
-                            setSelectedItems([path]);
-                          }}
-                        />
-                      )}
+                      {Object.values(metadataInstances).map((instance) => (
```

Comment:
why do we need to load into the DOM everything and not only the one that is currently selected?

Reply:
That's the trick to preserve the input's state for each opened node. Loading only the selected node is what causes the inputs to be 'reset' every time you switch from one node to another.

Reply:
but this is why we now have the instanceMetadata.data, no?
when we select a node, set also set the metadata data of this node (populate the relevant metadata fields)

Reply:
One thing is the metadata (node info) and the other the inputs the user fills in (for the api the request type, the endpoint...) 🤔.

Reply:
wouldn't it be a better solution to have a ResourceMetadataContext in our app that will have the user input for each resource? this way, once a resource is selected, its' user input can be fetched and displayed.
I think it will be better than to have all resources metadata panels and their content on the DOM

Reply:
If the component is removed from the DOM, we'd lose their state (which is what we really want to keep).

An alternative to this is to build a more complex set of hooks that preserve the state for us.